### PR TITLE
Show code instead of designer if the designer is not installed

### PR DESCRIPTION
**Pull request by sharwell**
See dotnet/winforms-designer#458

This avoids showing an error page when the designer package is not installed, but doesn't prevent the "View designer" command from showing in Solution Explorer (the command will just show code in this case).

### CODE REVIEW

File: src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs [hunk 1]
```diff
@@ -173,7 +177,7 @@ int IVsEditorFactory.Close()
             return VSConstants.S_OK;
         }
 
-        private static string GetWinFormsLoaderName(IVsHierarchy vsHierarchy)
+        private string GetWinFormsLoaderName(IVsHierarchy vsHierarchy)
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs]: Please annotate this as now being null returning, or rename to TryGet…

File: src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs [hunk 2]
```diff
@@ -193,6 +197,18 @@ private static string GetWinFormsLoaderName(IVsHierarchy vsHierarchy)
...
                     frameworkName.Version?.Major >= 3)
                 {
+                    if (!(_oleServiceProvider.QueryService<SVsShell>() is IVsShell shell))
+                    {
+                        return null;
+                    }
+
+                    var newWinFormsDesignerPackage = new Guid("c78ca057-cc29-421f-ad6d-3b0943debdfc");
+                    if (!ErrorHandler.Succeeded(shell.IsPackageInstalled(newWinFormsDesignerPackage, out var installed))
+                        || installed != 1)
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs]:
```suggestion
                        || installed == 0)
```

(not all COM trues are 1 so I'm just paranoid)

File: src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs (hunk 1 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs]: Done!

File: src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs (hunk 2 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs]: Amended the original commit to include this change.

**Pull request merged**